iohk-monitoring/src/Cardano/BM/Data/LogItem.lhs

```diff
@@ -191,20 +191,20 @@ LogStructured could also be:
 Payload of a |LogObject|:
 \begin{code}
 data LOContent a = LogMessage a
-                 | LogError Text
+                 | LogError !Text
                  | LogRepeats
                    { lrRepeats :: {-# UNPACK #-} !Int
                    , lrFirst   :: !(LogObject a)
                    , lrLast    :: !(LogObject a)
                    }
-                 | LogValue Text Measurable
-                 | LogStructured BS.ByteString
-                 | ObserveOpen CounterState
-                 | ObserveDiff CounterState
-                 | ObserveClose CounterState
+                 | LogValue !Text !Measurable
+                 | LogStructured !BS.ByteString
```

Comment:
I can imagine that this will affect the lazy construction of log messages.
for 'LogError' and the others more special constructors this might be important, but LogStructured is basically every message traced.
can you verify that tracing structured messages is still lazy?

Reply:
The bang pattern forces the evaluation of the first "constructor" (WHNF).
AFAIK, it should not force the evaluation of the whole `ByteString` (to NF), especially when it's lazy like this (http://hackage.haskell.org/package/bytestring-0.10.10.0/docs/src/Data.ByteString.Lazy.Internal.html#ByteString).